### ISSUE TITLE
Geometry_Engine: Area bug

### ISSUE THREAD

**Issue by michaldengusiak** [BHoM/BHoM_Engine#1198]
#### Description:

negative area

![image](https://user-images.githubusercontent.com/26113670/64791675-f7fab680-d56f-11e9-811f-4311ae41f6fe.png)

#### How to replicate:

run script

#### Test file(s):

[test file](https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/BHoM_Engine/Geometry_Engine/Geometry_Engine-Issue1198-AreaBug?csf=1&e=VhjKPm)

**Comment by pawelbaran** [comment]
Looks like an issue with numerical vs geometrical tolerance. Especially for high tolerances (e.g. 0.01), the REFTolerance goes up >1, which does not make sense. A working temp solution seems to be to use only the reduced row echelon form and make the REF tolerance not exceed 1:

```
public static double REFTolerance(this double[,] matrix, double tolerance = Tolerance.Distance)
        {
            int d1 = matrix.GetLength(0);
            int d2 = matrix.GetLength(1);
            double maxRowSum = 0;

            for (int i = 0; i < d1; i++)
            {
                double rowSum = 0;
                for (int j = 0; j < d2; j++)
                {
                    rowSum += Math.Abs(matrix[i, j]);
                }
                maxRowSum = Math.Max(maxRowSum, rowSum);
            }

            return Math.Min(tolerance * Math.Max(d1, d2) * maxRowSum, 1 - tolerance);
        }
```

This needs to be tested though.

**Comment by michaldengusiak** [comment]
@pawelbaran thanks for the great call will run some tests .. locally

**Comment by pawelbaran** [comment]
This has been resolved with #1200 and can be closed.
